kernel/core/maths/matfuncs.hpp

```diff
@@ -3567,6 +3567,174 @@ static Matrix matrixFilter(const MatFuncData& funcData, const MatFuncErrorInfo&
 }
 
 
+/////////////////////////////////////////////////
+/// \brief Function to perform a circular shift
+/// of the elements in a matrix by n rows.
+///
+/// \param funcData const MatFuncData&
+/// \param errorInfo const MatFuncErrorInfo&
+/// \return Matrix
+///
+/////////////////////////////////////////////////
+static Matrix circShiftRow(const MatFuncData& funcData, const MatFuncErrorInfo& errorInfo)
+{
+    // Inputs: funcData.mat1 funcData.nVal
+    std::vector<mu::value_type> matData = funcData.mat1.data();
+
+    // Circular shift downwards (negative values upwards)
+    // TODO: remove the unnecessary int casts

```

Comment:
What about storing `matData.size()` as `int`. That will resolve many of the uncessary casts.